### ISSUE TITLE
Rent expiry regions not resetting (as discussed)

### ISSUE THREAD

**Issue by frizzbee30** [NLthijs48/AreaShop#100]
Hi, please find attached the config as requested. Something else I have noticed, which may or may not be connected, a number of regions I set up as farm plots that appear to have been missed, although 'rent-able', I did see a warning about the regions, or no regions being attached, but haven't
caught the warning to paste since.
Wouldn't let me attach the zip so posted the link here

https://www.dropbox.com/sh/66c89a7b7ycu4hc/AABQOCdRtvHrp1X1Oc4PhQvUa?dl=0

Thanks

**Comment by frizzbee30** [comment]
we also seem to have lost about 300 prison cell rent regions, would the swap from WG 5 to WG 6 have had any impact? 

We have been using the plugins for a very long time, and it only seem that within the last 6 - 12 months we have some rent/buy issues?

Thanks  :D

**Comment by NLthijs48** [comment]
Thanks for the data.

First off, losing regions (defined as the WorldGuard region is still there but it is not added to AreaShop anymore) was possible in older versions of AreaShop. If a region would not find its WorldGuard region or world, then it would be removed. This could for example happen if Multiverse or WorldGuard fails to startup correctly. In the recent AreaShop version this has been fixed, and a warning is displayed in the console/log at startup. So checking for messages at startup is recommended (the message about regions come just after the 'Done' thing, and not when AreaShop is enabled). Also check if you have the latest AreaShop installed.

Then the next thing, the region files in the data that are rented by a player all seem fine (the end date is after today). Might it be that for the regions that did not get unrented, that those regions are actually not registered in AreaShop anymore? Note that the WorldGuard greeting/flags and the sign might still show like it would if it is rented, but that does not mean AreaShop still controls the region. You can check if the sign still reacts to leftclicking (should show the region info of AreaShop), and if `/as info region <regionName>` still shows the region info. In the numbering of the region files I see a bunch of gaps, so it looks to me that quite a lot of regions are not registered in AreaShop anymore.

Let me know if that would explain the issues you are having.

**Comment by frizzbee30** [comment]
Hi, thanks for the input. We are using the latest version of AS (just a bit of info)

It does look like some regions aren't handled by the plugin anymore, and the the failure to start issue would explain it. I've checked the expired regions and AS doesn't recognise them. We have obviously had a glitch, or probably an interrupted MySQL connection to the worldGuard tables on boot up (we did some time ago) which caused an issue with some of the AS rents.

Thank you for taking the time to look into it, we can look at adding the regions back in.
